community/values/src/main/java/org/neo4j/values/storable/TemporalValue.java

```diff
@@ -393,8 +396,8 @@ public String toString()
         VALUE result = matcher.matches() ? parser.apply( matcher ) : null;
         if ( result == null )
         {
-            throw new DateTimeParseException(
-                    "Text cannot be parsed to a " + valueName( type ), text, 0 );
+            throw new TemporalParseException(
+                    "Text cannot be parsed to a " + valueName( type ), text.toString(), 0 );
```

Comment:
Sometimes you use `text.toString()` and sometimes `text.stringValue()`.

Reply:
That is because `text` has different types in the different methods (and had already before I touched it). Sometimes it is a `CharSequence` and sometimes a `TextValue`